lib/Parse/ParseType.cpp

```diff
@@ -1682,13 +1677,9 @@ bool Parser::canParseTypeIdentifier() {
 bool Parser::canParseBaseTypeForQualifiedDeclName() {
   BacktrackingScope backtrack(*this);
 
-  // Parse a simple type identifier.
-  if (!canParseSimpleTypeIdentifier())
-    return false;
-
   // Qualified name base types must be followed by a period.
   // If the next token starts with a period, return true.
-  return startsWithSymbol(Tok, '.');
+  return (canParseSimpleTypeIdentifier() && startsWithSymbol(Tok, '.'));
```

Comment:
I prefer the original. `if (!canParseSimpleTypeIdentifier()) return false;` is more like an early return than combined condition.